doc/hacking.rst

```diff
@@ -779,46 +779,115 @@ https://ocaml.github.io/dune/dev/bench/
-Build the Docker image from the root directory of this repo.
+The Monorepo Benchmark is designed to assess the performance of Dune in
+building a large monorepo, specifically constructed using `opam-monorepo
+<https://github.com/tarides/opam-monorepo>`_.
 
-E.g., run
+Current Monorepo Benchmarks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. code:: console
+TODO
 
-   $ docker build . -f bench/monorepo/bench.Dockerfile --tag=dune-monorepo-benchmark
+Adding new benchmarks
+^^^^^^^^^^^^^^^^^^^^^
 
-The monorepo benchmark Docker image requires ``duniverse`` directory to be mounted
-as a volume. Generate this directory with a script from the
-`ocaml-monorepo-benchmark <https://github.com/ocaml-dune/ocaml-monorepo-benchmark>`_
-repository:
+TODO
 
-.. code:: console
+Setting Up the Benchmark Environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-   $ git clone https://github.com/ocaml-dune/ocaml-monorepo-benchmark.git
-   $ cd ocaml-monorepo-benchmark
-   $ ./generate-duniverse.sh /tmp
+To set up the Docker environment for the Monorepo Benchmark, you will need to
+build a Docker image and generate and mount a ``duniverse`` directory.
 
-This will create a directory ``/tmp/duniverse``. Then to run the benchmark, run
-the Docker image in a container mounting ``/tmp/duniverse`` as a volume at
-``/home/opam/bench-dir/current-bench-data/duniverse`` (that specific path is a
-requirement of `current-bench <https://github.com/ocurrent/current-bench>`_).
-From within the container the benchmarks can be started by running `make bench`.
-Do all this in a single command with:
+Building the Docker Image
+^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. code:: console
+#. **Build the Docker Image**: From the root directory of your local Dune
+   repository, build the Docker image using the
+   ``bench/monorepo/bench.Dockerfile``
+
+   .. code:: console
+
+      $ docker build . -f bench/monorepo/bench.Dockerfile --tag=dune-monorepo-benchmark
+
+Preparing the ``duniverse`` Directory
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. **Clone the Benchmark Repository**: Clone the ``ocaml-monorepo-benchmark``
+   repository to obtain necessary scripts and files.
+
+   .. code:: console
+
+      $ git clone https://github.com/ocaml-dune/ocaml-monorepo-benchmark.git
+      $ cd ocaml-monorepo-benchmark
+
+#. **Generate the duniverse Directory**: Use the ``generate-duniverse.sh``
+   script to create the ``duniverse`` directory at a desired location (e.g.,
+   ``/tmp``).
+
+   .. code:: console
+
+      $ ./generate-duniverse.sh /tmp
+
+Running the Benchmark
+~~~~~~~~~~~~~~~~~~~~
+
+Once the Docker image is built and the ``duniverse`` directory is prepared, you
+can run the benchmark.
+
+#. **Run the Benchmark**: Execute the Docker container, mounting the
+   ``duniverse`` directory as a volume.
+
+   .. code:: console
+
+      $ docker run -it --volume=/tmp/duniverse:/home/opam/bench-dir/current-bench-data/duniverse dune-monorepo-benchmark bash --login -c 'make bench'
+
+
+Understanding the Results
```

Comment:
The previous sections give instructions on how to run the benchmarks locally but the remainder of the doc is about current-bench. Is there a risk that this suggests that you can use current-bench to analyze results obtained by running the benchmark locally? Maybe mention that current-bench is only available when running the benchmark remotely and the only reason one would run the benchmarks locally is for debugging.